modules/structs/repo.go

```diff
@@ -160,6 +160,23 @@ type EditRepoOption struct {
 	Archived *bool `json:"archived,omitempty"`
 }
 
+// CreateBranchRepoOption options when creating a branch in a repository
+// swagger:model
+type CreateBranchRepoOption struct {
+
+	// Name of the branch to create
+	//
+	// required: true
```

Comment:
I would make it Optional and use DefaultBranch on ""

Reply:
else it would not make sence and I would use
`m.Post("/*", reqRepoWriter(models.UnitTypeCode), context.RepoRefByType(context.RepoRefBranch), repo.CreateBranch)`

Reply:
Sory I mean "OldBranchName" with this ...

Reply:
I made the DefaultBranch optional and added the reqRepoWriter.

Since the arguments (new_branch_name and old_branch_name) are in the json parameters, is it necessary to add an url parameter (```"/*"```) ?

```
m.Post("", reqRepoWriter(models.UnitTypeCode), bind(api.CreateBranchRepoOption{}), repo.CreateBranch)
```

Reply:
No, it was an either ... or

Reply:
Ok, are you good with this solution ?

```m.Post("", reqRepoWriter(models.UnitTypeCode), bind(api.CreateBranchRepoOption{}), repo.CreateBranch)```

Reply:
since you squashed it all, I'll review the hole thing